compiler/rustc_infer/src/traits/mod.rs

```diff
@@ -144,7 +144,7 @@ pub enum FulfillmentErrorCode<'tcx> {
     Ambiguity {
         /// Overflow reported from the new solver `-Znext-solver`, which will
         /// be reported as an regular error as opposed to a fatal error.
-        overflow: bool,
+        overflow: Option<bool>,
```

Comment:
Could you expand the comment above on what this `Option<bool>` means (or use an enum for the bool) (from skimming over this PR, I think the bool means "suggest recursion limit" now?)

Reply:
Yes pls

Reply:
done